.ci/cico_olm_openshift.sh

```diff
@@ -1,72 +1,100 @@
 #!/bin/bash
 #
-# Copyright (c) 2012-2020 Red Hat, Inc.
+# Copyright (c) 2020 Red Hat, Inc.
 # This program and the accompanying materials are made
 # available under the terms of the Eclipse Public License 2.0
 # which is available at https://www.eclipse.org/legal/epl-2.0/
 #
 # SPDX-License-Identifier: EPL-2.0
 #
+# Contributors:
+#   Red Hat, Inc. - initial API and implementation
+
+set -ex
+
+#Stop execution on any error
+trap "catchFinish" EXIT SIGINT
+
+# Catch_Finish is executed after finish script.
+catchFinish() {
+  result=$?
+  if [ "$result" != "0" ]; then
+    echo "Failed on running tests. Please check logs or contact QE team (e-mail:codereadyqe-workspaces-qe@redhat.com, Slack: #che-qe-internal, Eclipse mattermost: 'Eclipse Che QE'"
+    echo "Logs should be availabe on http://artifacts.ci.centos.org/devtools/che/che-eclipse-minikube-updates/${ghprbPullId}/"
+    exit 1
+    getCheClusterLogs
+    archiveArtifacts "che-operator-minikube-updates"
+  fi
+  rm -rf ~/.kube ~/.minikube
+  exit $result
+}
 
-set -e
-
-# Detect the base directory where che-operator is cloned
-SCRIPT=$(readlink -f "$0")
-export SCRIPT
-
-OPERATOR_REPO=$(dirname "$(dirname "$SCRIPT")");
-export OPERATOR_REPO
-
-# ENV used by Openshift CI
-ARTIFACTS_DIR="/tmp/artifacts"
-export ARTIFACTS_DIR
-
-# Component is defined in Openshift CI job configuration. See: https://github.com/openshift/release/blob/master/ci-operator/config/devfile/devworkspace-operator/devfile-devworkspace-operator-master__v4.yaml#L8
-CI_COMPONENT="che-operator-catalog"
-export CI_COMPONENT
-
-CATALOG_SOURCE_IMAGE_NAME=${CI_COMPONENT}:stable
-export CATALOG_SOURCE_IMAGE_NAME
-
-# This image is builded by Openshift CI and exposed to be consumed for olm tests.
-#OPENSHIFT_BUILD_NAMESPACE env var exposed by Openshift CI. More info about how images are builded in Openshift CI: https://github.com/openshift/ci-tools/blob/master/TEMPLATES.md#parameters-available-to-templates
-CATALOG_SOURCE_IMAGE=registry.svc.ci.openshift.org/${OPENSHIFT_BUILD_NAMESPACE}/${CATALOG_SOURCE_IMAGE_NAME}
-export CATALOG_SOURCE_IMAGE
+init() {
+  SCRIPT=$(readlink -f "$0")
+  SCRIPT_DIR=$(dirname "$SCRIPT")
 
-# Choose if install Eclipse Che using an operatorsource or Custom Catalog Source
-INSTALLATION_TYPE="catalog"
-export INSTALLATION_TYPE
+  if [[ ${WORKSPACE} ]] && [[ -d ${WORKSPACE} ]]; then
+    OPERATOR_REPO=${WORKSPACE};
+  else
+    OPERATOR_REPO=$(dirname "$SCRIPT_DIR");
+  fi
 
-# Execute olm nightly files in openshift
-PLATFORM="openshift"
-export PLATFORM
+  RAM_MEMORY=8192
+  PLATFORM="kubernetes"
```

Comment:
Strange, I thought that test cico_olm_openshift.sh is about openshift and should be executed on the openshift CI...

Reply:
@AndrienkoAleksandr my mistake